R/ihw_class.R

```diff
@@ -161,13 +161,10 @@ setMethod("weighted_pvalues", signature(object="ihwResult"),
 
 #---------------------------  covariate extraction ----------------------------------------------------------#
 
-covariates_ihwResult <- function(object){
-  covariate <- dplyr::select(object@df, dplyr::starts_with("covariate")) #TODO fix this
-  if(ncol(covariate) == 1){
-    unlist(covariate)
-  }else{
-    as.matrix(covariate)
-  }
+covariates_ihwResult <- function(object){ 
+  covariate_names <- grepl("^covariate_", colnames(object@df))
```

Comment:
Sorry that was probably a bad example. The underscore `_` had no particular meaning in that regex!